model/src/main/kotlin/io/spine/validation/bound/NumericBoundParser.kt

```diff
@@ -0,0 +1,201 @@
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.validation.bound
+
+import io.spine.protodata.Compilation
+import io.spine.protodata.ast.PrimitiveType
+import io.spine.protodata.ast.PrimitiveType.TYPE_DOUBLE
+import io.spine.protodata.ast.PrimitiveType.TYPE_FIXED32
+import io.spine.protodata.ast.PrimitiveType.TYPE_FIXED64
+import io.spine.protodata.ast.PrimitiveType.TYPE_FLOAT
+import io.spine.protodata.ast.PrimitiveType.TYPE_INT32
+import io.spine.protodata.ast.PrimitiveType.TYPE_INT64
+import io.spine.protodata.ast.PrimitiveType.TYPE_SFIXED32
+import io.spine.protodata.ast.PrimitiveType.TYPE_SFIXED64
+import io.spine.protodata.ast.PrimitiveType.TYPE_SINT32
+import io.spine.protodata.ast.PrimitiveType.TYPE_SINT64
+import io.spine.protodata.ast.PrimitiveType.TYPE_UINT32
+import io.spine.protodata.ast.PrimitiveType.TYPE_UINT64
+import io.spine.protodata.ast.name
+import io.spine.protodata.ast.qualifiedName
+import io.spine.protodata.check
+import io.spine.protodata.type.resolve
+import io.spine.validation.bound.BoundFieldSupport.numericPrimitives
+
+/**
+ * Parses bound values specified in Protobuf options that restrict
+ * the value range of numeric fields.
+ *
+ * The parser supports both number literals and field references.
+ *
+ * @param metadata The information about the numeric option and the field it constrains,
+ *   used to interpret and validate the bound value.
+ */
+internal class NumericBoundParser(
+    private val metadata: NumericOptionMetadata
+) {
+
+    /**
+     * Parses the given raw [bound] value to a [KNumericBound].
+     *
+     * For number-based bounds, the method checks the following:
+     *
+     * 1) The bound value is not empty.
+     * 2) The provided number has `.` for floating-point fields, and does not have `.`
+     *    for integer fields.
+     * 3) The provided number fits into the range of the target field type.
+     *
+     * For field-based bounds:
+     *
+     * 1) The specified field path points to an existing field.
+     * 2) The field bound is not referencing the field it restricts (self-referencing).
+     * 3) The referenced field is of singular numeric type.
+     *
+     * Any violation of the above conditions leads to a compilation error.
+     *
+     * @return The parsed numeric bound.
+     */
+    internal fun parse(bound: String, exclusive: Boolean): KNumericBound {
+        with(metadata) {
+            Compilation.check(bound.isNotEmpty(), file, field.span) {
+                "The `($optionName)` option could not parse the bound value specified for" +
+                        " `${field.qualifiedName}` field because it is empty. Please provide" +
```

Comment:
Please have the ` Please provide` staring on the new line in the code, and in the console output.
This way it's easier to read.